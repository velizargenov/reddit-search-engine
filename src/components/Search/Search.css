input {
  padding: 6px;
  margin-top: 8px;
  font-size: 17px;
}
button {
  border: 0;
  height: 36px;
  max-height: 33px;
  font-size: 22px;
  cursor: pointer;
}