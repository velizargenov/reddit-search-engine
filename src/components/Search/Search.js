import React from 'react';
import PropTypes from 'prop-types';

import './Search.css';

const Search = ({ handleSubmit, handleChange, value }) => (
  <form className='search' onSubmit={handleSubmit}>
    <input type="text" value={value} onChange={handleChange} />
    <input type="submit" value="Search" />
  </form>
)

Search.propTypes = {
  value: PropTypes.string,
  handleSubmit: PropTypes.func,
  handleChange: PropTypes.func
}

export default Search;
