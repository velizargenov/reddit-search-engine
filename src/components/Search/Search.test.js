import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow, mount } from 'enzyme';

import Search from './Search';

describe('about Search', () => {
  let wrapper;

  describe('about rendering', () => {
    beforeEach(() => {
      wrapper = shallow(<Search />)
    });

    it('should render component without crashing', () => {
      expect(wrapper.length).toBe(1);
    });

    it('matches snapshot', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });

  it('calls onSubmit when the form is submitted', () => {
    const onSubmitFn = jest.fn();
    wrapper = mount(<Search handleSubmit={onSubmitFn} />);

    const form = wrapper.find('form');
    form.simulate('submit');
    expect(onSubmitFn).toHaveBeenCalledTimes(1);
  });

});
