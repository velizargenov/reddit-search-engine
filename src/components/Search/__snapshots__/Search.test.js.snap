// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`about Search about rendering matches snapshot 1`] = `
<form
  className="search"
>
  <input
    type="text"
  />
  <input
    type="submit"
    value="Search"
  />
</form>
`;
